eng/pipelines/templates/jobs/ci.tests.yml

```diff
@@ -152,15 +152,15 @@ jobs:
           runProxy: true
           rootFolder: '$(Build.SourcesDirectory)/sdk/${{parameters.ServiceDirectory}}'
           templateFolder: '$(Build.SourcesDirectory)/sdk/${{parameters.ServiceDirectory}}'
-          condition: and(succeeded(), contains(variables.CmakeArgs, 'BUILD_TESTING=ON'))
+          condition: and(succeeded(), contains(variables.CmakeArgs, 'BUILD_TESTING=ON'),ne('${{parameters.ServiceDirectory}}', 'template'))
```

Comment:
I would prefer to have someone from EngSys look at this change, before merging.

Reply:
Why do we need to exclude the template package from test proxy? And if we really need to, is this the cleanest way to do it?

We are adding exception in multiple places, which seems a bit fragile. Are these the only places that need to opt-out?

Reply:
@gearama @ahsonkhan we also depend on the template pipeline to validate changes, often ones we make that affect all repos. I would prefer not to special case template so that we can catch any possible common proxy setup issues that might affect c++.